skyportal/models/schema.py

```diff
@@ -1036,6 +1036,20 @@ class SpectrumAsciiFilePostJSON(SpectrumAsciiFileParseJSON):
         description='The ID of the instrument that took the spectrum.', required=True
     )
 
+    # type = ApispecEnumField(
```

Comment:
Should probably be:

```
ApispecEnumField(
    Enum('type', py_allowed_spectrum_types), required=True/False
)
```

Reply:
Tried this. App doesn't start up. Tried replacing `py_allowed_spectrum_types` -> `ALLOWED_SPECTRUM_TYPES` and now I get errors when posting spectra (`'type.host' is not among the defined enum values. Enum name: spectrumtypes. Possible values: source, host, host_center`)

Reply:
This is the error I get when I use `py_allowed_spectrum_types`: 
`Traceback (most recent call last):
  File "baselayer/services/app/app.py", line 36, in <module>
    app_factory = getattr(importlib.import_module(module), app_factory)
  File "/home/guyn/anaconda3/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/guyn/skyportal/skyportal/app_server.py", line 7, in <module>
    from skyportal.handlers import BecomeUserHandler, LogoutHandler
  File "/home/guyn/skyportal/skyportal/handlers/__init__.py", line 11, in <module>
    from .become_user import BecomeUserHandler
  File "/home/guyn/skyportal/skyportal/handlers/become_user.py", line 2, in <module>
    from ..models import User
  File "/home/guyn/skyportal/skyportal/models/__init__.py", line 40, in <module>
    from .schema import setup_schema
  File "/home/guyn/skyportal/skyportal/models/schema.py", line 1031, in <module>
    class SpectrumAsciiFilePostJSON(SpectrumAsciiFileParseJSON):
  File "/home/guyn/skyportal/skyportal/models/schema.py", line 1041, in SpectrumAsciiFilePostJSON
    Enum('type', py_allowed_spectrum_types),
  File "/home/guyn/anaconda3/lib/python3.8/enum.py", line 341, in __call__
    return cls._create_(
  File "/home/guyn/anaconda3/lib/python3.8/enum.py", line 463, in _create_
    member_name, member_value = item
TypeError: cannot unpack non-iterable spectrumtypes object`

Reply:
That error message seems correct.  Why should the enum allow the value of `host.name`?

Reply:
I am posting with `type: 'host"` but that is translated to `type.host` and that isn't allowed...

Reply:
Do you know where that translation happens and why?

Reply:
No, I just tried to do what is done for other classes and this is what happened...

Reply:
![image](https://user-images.githubusercontent.com/45071/140264290-b5e638d6-5bf1-41c0-b8c6-d367acb9f9f4.png)